Src/Support/IntegrationTests/PerCallAuthTests.cs

```diff
@@ -61,6 +64,151 @@ public void OverrideServiceCredentialWithPerCallCredential()
+
+            // Set one credential in the client
+            StorageService client = new StorageService(new BaseClientService.Initializer
+            {
+                ApplicationName = "IntegrationTest",
+                HttpClientInitializer = serviceCredential
+            }); ;
+
+            // Create a request with a per-call credential.; this overrides the service-level credential.
+            var request = client.Buckets.List(Helper.GetProjectId()).AddCredential(callCredential);
+
+            // This should throw unauthorized, the credentials are fake, but let's make sure the
+            // request got to the server so we know all appropiate interceptors executed.
+            var exception = Assert.ThrowsAny<GoogleApiException>(() => request.Execute());
+            Assert.Equal(401, exception.Error.Code);
+
+            HttpRequestMessage httpRequest = serviceCredential.Request;
+            string query = httpRequest.RequestUri.Query;
+
+            int serviceIndex = query.LastIndexOf("access_token=SERVICE_CREDENTIAL_TOKEN", StringComparison.OrdinalIgnoreCase);
+
+            Assert.Equal(-1, serviceIndex);
+            Assert.Equal("CALL_CREDENTIAL_TOKEN", httpRequest.Headers.Authorization.Parameter);
+        }
+
+        [Fact]
+        public void OverridesServiceCredentialWithPerCallCredential_TwoCallCredentials()
+        {
+            var serviceCredential = new RequestInterceptorCredentialWrapper("SERVICE_CREDENTIAL_TOKEN", new BearerToken.QueryParameterAccessMethod());
+            var callCredential1 = new AccessTokenCredential("CALL_CREDENTIAL_TOKEN_1", new BearerToken.QueryParameterAccessMethod());
+            var callCredential2 = new AccessTokenCredential("CALL_CREDENTIAL_TOKEN_2", new BearerToken.QueryParameterAccessMethod());
+
+            // Set one credential in the client
+            StorageService client = new StorageService(new BaseClientService.Initializer
+            {
+                ApplicationName = "IntegrationTest",
+                HttpClientInitializer = serviceCredential
+            }); ;
+
+            // Create a request with a per-call credential.; this overrides the service-level credential.
+            var request = client.Buckets.List(Helper.GetProjectId()).AddCredential(callCredential1);
+            request.AddCredential(callCredential2);
+
+            // This should throw unauthorized, the credentials are fake, but let's make sure the
+            // request got to the server so we know all appropiate interceptors executed.
+            var exception = Assert.ThrowsAny<GoogleApiException>(() => request.Execute());
+            Assert.Equal(401, exception.Error.Code);
+
+            string query = serviceCredential.Request.RequestUri.Query;
+
+            int serviceIndex = query.IndexOf("access_token=SERVICE_CREDENTIAL_TOKEN", StringComparison.OrdinalIgnoreCase);
```

Comment:
Maybe helper methods of "AssertQueryDoesNotContainToken" and "AssertQueryContainsToken" (and maybe ditto for headers) accepting an HttpRequest and a string would be useful for clarifying these tests. Or maybe it would be overkill. I think it would make the tests read super-easily though :)